core/trino-main/src/main/java/io/trino/sql/analyzer/FeaturesConfig.java

```diff
@@ -80,6 +80,7 @@
     private int maxReorderedJoins = 9;
     private boolean redistributeWrites = true;
     private boolean usePreferredWritePartitioning;
```

Comment:
add a separate commit that enables `usePreferredWritePartitioning` by default. I think we should atomically change semantics (instead of two PRs) and describe changes in release notes.